### ISSUE TITLE
Fix CI notebook tests for MacOS and Windows

### ISSUE THREAD

**Issue by pavoljuhas** [quantumlib/Cirq#6336]
**Description of the issue**

After activating MacOS and Windows testing in [ci-daily.yml](https://github.com/quantumlib/Cirq/blob/master/.github/workflows/ci-daily.yml) in #6331, the notebook tests failed on these platforms, [example](https://github.com/quantumlib/Cirq/actions/runs/6700241980).
As a temporary solution, #6335 restricts notebook testing to run on Linux only.
For a proper solution we need to fix notebook tests on Mac and Windows platforms.

The affected notebook tests are those touched in #6335, ie, those decorated with
``` 
@pytest.mark.skipif(sys.platform != "linux", reason="Linux-only test")
```

**Cirq version**
1.3.0.dev at 34e8dab087c65ff62957e8fc33c418f19f47333a

**Comment by smburdick** [comment]
I can take this on.

I can use my own local Windows/MacOS environments to reproduce the issues.

**Comment by smburdick** [comment]
https://github.com/quantumlib/Cirq/commit/3c6a305bb88b93b197c8c98762013baa1134f41e removed all but one of the `skipif` tags.

The remaining one is in `dev_tools/notebooks/notebook_test.py`.

Removing that line and running `pytest` on macOS makes the test pass. Should this still be open for mac? I will try it on Windows later.

```
$ ./check/pytest-changed-files
Comparing against revision 'upstream/main'.
Found 1 test files associated with changes.
============================================================================================= test session starts ==============================================================================================
platform darwin -- Python 3.11.2, pytest-7.4.3, pluggy-1.3.0
Using --randomly-seed=387391441
rootdir: /Users/sam/Cirq
configfile: pyproject.toml
plugins: randomly-3.15.0, asyncio-0.21.1, xdist-2.2.1, anyio-4.1.0, forked-1.6.0, cov-3.0.0
asyncio: mode=Mode.STRICT
collected 56 items

dev_tools/notebooks/notebook_test.py ........................................................                                                                                                            [100%]

======================================================================================== 56 passed in 660.27s (0:11:00) ========================================================================================
```

**Comment by smburdick** [comment]
From the macOS output:
```
dev_tools/notebooks/notebook_test.py:123: Failed
----------------------------- Captured stdout call -----------------------------
/bin/sh: line 1: papermill: command not found
```

Looks like we need to add `papermill` to the requirements list

**Comment by smburdick** [comment]
Windows output:
```
-r: D:\a\_temp\c0ad28f8-cc4a-442c-b165-389fe89d7f5e.ps1:4
Line |
   4 |    -r dev_tools/requirements/deps/format.txt \
     |    ~~
     | The term '-r' is not recognized as a name of a cmdlet, function, script file, or executable program. Check the
     | spelling of the name, or if a path was included, verify that the path is correct and try again.

Error: Process completed with exit code 1.
##[debug]Finishing: Install requirements
```
I'm not too well versed on Windows/PowerShell, but it looks like you're trying to run a Unix style multiline command in a Powershell/cmd context.

**Comment by smburdick** [comment]
https://github.com/quantumlib/Cirq/commit/30b6c3908e36c58299e920f03803c0f9d8d23668 addresses macOS bug. Will begin on Windows side now.

**Comment by smburdick** [comment]
I'm a little out of my depth on the Windows side of things (all my dev experience is Unix). Please unassign me, and I will pick up something else. Thanks!

**Comment by pavoljuhas** [comment]
Based on the test runs in #7479, the execution of notebook tests assumes a POSIX-like environment, for example, temporary virtual environments are activated with a [shell syntax](https://github.com/quantumlib/Cirq/blob/89ebad8ca3c5a742d40e459fd4387cf357f2a0c5/dev_tools/notebooks/isolated_notebook_test.py#L154-L156).  Users are likely to execute notebooks in Colabs which come with a unix-like engine.

In summary, it is not worth the effort to get notebook tests running on Windows.  (They already run and test on Mac as of #6366).

Closing as wontfix.  (reopen if we get compelling reasons to test on Windows)
